SMLHelper/Handlers/TechTypeHandler.cs

```diff
@@ -60,7 +61,30 @@ TechType ITechTypeHandlerInternal.AddTechType(string modName, string internalNam
         #region Static Methods
 
         /// <summary>
-        /// Adds a new <see cref="TechType"/> into the game. This new techtype will be unlocked at the start of a the game.
+        /// Adds a new <see cref="TechType"/> into the game. This new <see cref="TechType"/> will be unlocked at the start of a game.
+        /// </summary>
+        /// <param name="addedByAssembly">The assembly this TechType is getting added by.</param>
+        /// <param name="internalName">The internal name of the TechType. Should not contain special characters</param>
+        /// <param name="displayName">The display name of the TechType. Can be anything.</param>
+        /// <param name="tooltip">The tooltip, displayed when hovered in an inventory. Can be anything.</param>
+        /// <returns>The new <see cref="TechType"/> that is created.</returns>
+        public static TechType AddTechType(Assembly addedByAssembly, string internalName, string displayName, string tooltip) =>
+            Singleton.AddTechType(addedByAssembly, internalName, displayName, tooltip);
+
+        /// <summary>
+        /// Adds a new <see cref="TechType"/> into the game.
+        /// </summary>
+        /// <param name="addedByAssembly">The assembly this TechType is getting added by.</param>
+        /// <param name="internalName">The internal name of the TechType. Should not contain special characters</param>
+        /// <param name="displayName">The display name of the TechType. Can be anything.</param>
+        /// <param name="tooltip">The tooltip, displayed when hovered in an inventory. Can be anything.</param>
+        /// <param name="unlockAtStart">Whether this TechType should be unlocked on game start, or not. By default, <see langword="true"/>.</param>
+        /// <returns>The new <see cref="TechType"/> that is created.</returns>
+        public static TechType AddTechType(Assembly addedByAssembly, string internalName, string displayName, string tooltip, bool unlockAtStart) =>
```

Comment:
So how is this going to solve the issue?  
Because it's not "wrong". SMLHelper is recording which assembly made the request to it. And if that assembly was a library and not a final mod, that's just what happened.  
It's why all items created by CustomBatteries just report that mod as their origin.

Does this require other mods to pass in the assembly of the calling mod? If so, you've only "fixed" this for the libraries where the author knows to do that.

Reply:
SMLHelper is indeed just doing what it has been told to do and there's nothing to blame it about, but before there was no way to "fix" this kind of issues for library mods and now there is.

And pretty much the only other libraries that messes with techtype adding are ECCLibrary and the upcoming ArchitectsLibrary - both are actively being developed on and aware of this issue.

Libraries do indeed need to pass the assembly manually if they encounter to this issue and normally this issue only happens if the techtype wasnt made by a Spawnable class or another mod's assets class that inherits from SMLHelper ones, its mostly just for extra techtypes made manually with TechTypeHandler.AddTechType().

If u have any better workaround on top of ur head, let me know :)

Reply:
I find it surprising that we still have to manually invoke `AddTechType` at all.  
But that aside, as long as this is a _new_ method being added and the original (public) ones are still in place and with their original signatures, then we're good.